tests/kafkatest/tests/client/client_compatibility_features_test.py

```diff
@@ -107,29 +104,7 @@ def invoke_compatibility_program(self, features):
 
     @cluster(num_nodes=7)
     @matrix(broker_version=[str(DEV_BRANCH)], metadata_quorum=quorum.all_non_upgrade)
```

Comment:
Don't we want to add brroker versions going back to 3.4 here? That was the first GA version of KRaft.

(Upgrardes from 3.3 and later to 4.x are fine and supported, as long as they're KRaft upgrades)